proxy/pkg/zdmproxy/pscache.go

```diff
@@ -64,31 +81,37 @@ func (psc *PreparedStatementCache) StoreIntercepted(preparedResult *message.Prep
 func (psc *PreparedStatementCache) Get(originPreparedId []byte) (PreparedData, bool) {
 	psc.lock.RLock()
 	defer psc.lock.RUnlock()
```

Comment:
We need to reevaluate these locks. This new cache implementation already does locking so it would be great if we could remove our locks.